Assets/MRTK/SDK/Experimental/SpatialAwareness/SurfaceMeshesToPlanes.cs

```diff
@@ -0,0 +1,487 @@
+
+        [SerializeField]
+        [Tooltip("Buffer to use when determining if a horizontal plane near the ceiling should be considered part of the ceiling.")]
+        [Range(0.0f, 1.0f)]
+        private float ceilingBuffer = 0.1f;
+
+        /// <summary>
+        /// Buffer to use when determining if a horizontal plane near the ceiling should be considered part of the ceiling.
+        /// </summary>
+        public float CeilingBuffer
+        {
+            get => ceilingBuffer;
+            set => ceilingBuffer = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Thickness of rendered plane objects")]
+        [Range(0.0f, 1.0f)]
+        private float planeThickness = 0.01f;
+
+        /// <summary>
+        /// Thickness of rendered plane objects
+        /// </summary>
+        public float PlaneThickness
+        {
+            get => planeThickness;
+            set => planeThickness = value;
+        }
+
+        [HideInInspector]
+        private SpatialAwarenessSurfaceTypes drawPlanesMask =
+            (SpatialAwarenessSurfaceTypes.Wall | SpatialAwarenessSurfaceTypes.Floor | SpatialAwarenessSurfaceTypes.Ceiling | SpatialAwarenessSurfaceTypes.Platform);
+
+        /// <summary>
+        /// Determines which plane types should be rendered
+        /// </summary>
+        public SpatialAwarenessSurfaceTypes DrawPlanesMask
+        {
+            get => drawPlanesMask;
+            set => drawPlanesMask = value;
+        }
+
+        [HideInInspector]
+        private SpatialAwarenessSurfaceTypes destroyPlanesMask = SpatialAwarenessSurfaceTypes.Unknown;
+
+        /// <summary>
+        /// Determines which plane types should be discarded.
+        /// Use this when the spatial mapping mesh is a better fit for the surface (ex: round tables).
+        /// </summary>
+        public SpatialAwarenessSurfaceTypes DestroyPlanesMask
+        {
+            get => destroyPlanesMask;
+            set => destroyPlanesMask = value;
+        }
+
+        /// <summary>
+        /// Delegate which is called when the MakePlanesCompleted event is triggered.
+        /// </summary>
+        public delegate void EventHandler(object source, EventArgs args);
+
+#if PLANE_FINDING_PRESENT
+
+        /// <summary>
+        /// EventHandler which is triggered when the MakePlanesRoutine is finished.
+        /// </summary>
+        public event EventHandler MakePlanesComplete;
+
+        /// <summary>
+        /// Indicates whether or not the project contains the required components for SurfaceMeshesToPlanes
+        /// to enable plane creation.
+        /// </summary>
+        /// <remarks>
+        /// For SurfaceMeshesToPlanes to create planes, the Mixed Reality Toolkit Plane Finding package
+        /// must be imported.
+        /// </remarks>
+        public static bool CanCreatePlanes =>
+#if PLANE_FINDING_PRESENT
+            true;
+#else
+            false;
+#endif // PLANE_FINDING_PRESENT
+
+
+#endif // PLANE_FINDING_PRESENT
```

Comment:
This is inside an `#if PLANE_FINDING_PRESENT` on line 169. I'll move it out in my upcoming commit